processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -195,6 +196,28 @@ public static ImmutableBitmap allTrue(final BitmapIndexSelector selector)
   }
 
   /**
+   * Union an iterable of bitmaps.
+   *
+   * @param bitmapFactory factory corresponding to the bitmaps
+   * @param bitmaps       the bitmaps
+   *
+   * @return unioned bitmap
+   */
+  public static ImmutableBitmap union(final BitmapFactory bitmapFactory, final Iterable<ImmutableBitmap> bitmaps)
```

Comment:
This logic should just be part of every `BitmapFactory.union()` implementation. No ambiguity, impossible to choose "wrong" method and miss some optimization.